### PR TITLE
Adds benches for Serpent + questions regarding optimization

### PR DESCRIPTION
**Pull request by Slals**
I added benches for Serpent but results are not so satisfying, it could be x3-x10 better.

Here are the benchmark results :

```
running 2 tests
test decrypt ... bench:       8,447 ns/iter (+/- 200) = 1 MB/s
test encrypt ... bench:       8,325 ns/iter (+/- 676) = 1 MB/s
```

I have one simple question : could you guys provide some hints to improve the current Serpent implementation encrypt and decrypt speed? I don't mean Serpent specifics but more like Rust performance optimization such as `inline` and good practices.

Another question for you @tarcieri, earlier you told me about checking `branch instruction` on ASM, since I'm kinda newbie on this could you tell me what I have to consider regarding this?

Thanks!

### CODE REVIEW

**Comment by tarcieri** [comment]
@Slals I don't know much about Serpent, however those numbers certainly do seem quite slow

File: serpent/src/lib.rs [hunk 1]
```diff
@@ -229,6 +227,7 @@ impl BlockCipher for Serpent {
         Self::new_varkey(key).unwrap()
     }
 
+    #[inline(always)]
```
Comment (tarcieri) [inline serpent/src/lib.rs]: Did these actually help? IMO every `#[inline]` directive should have a comment explaining what it's doing and why that's better than what the LLVM optimizer did by default

**Comment by jack-signal** [comment]
Right now it seems like Serpent is implemented in this crate using table lookups instead of the usual bitsliced sbox expressions. Changing that would probably provide about a 50x-100x performance improvement. In bitsliced mode it is also easy to use SIMD for parallel execution.

**Comment by tarcieri** [comment]
@Slals would you mind changing this PR to just include the benchmarks for now? Thanks!

**Comment by Slals** [comment]
Thanks @jack-signal for your feedback. I did't take the time to make the change in order to speed it. 

@tarcieri I removed the "inlines" tags if it's what you needed.

**Comment by tarcieri** [comment]
@Slals there are some conflicting changes in `serpent/src/lib.rs`. They might just go away if you run `rustfmt` on this branch. Otherwise you need to rebase.

**Comment by tarcieri** [comment]
I think the original goal of this PR was actually accomplished in #105 last May. Going to close this out.

**Pull request closed without merging**